code/modules/arcade/mob_hunt/mob_avatar.dm

```diff
@@ -0,0 +1,162 @@
+
+/obj/effect/nanomob
+	name = "Nano-Mob Avatar"					//will be overridden by the mob datum name value when created
+	desc = "A wild Nano-Mob appeared! Hit it with your PDA with the game open to attempt to capture it!"
+	invisibility = 101
+	alpha = 128
+	anchored = 1								//just in case
+	density = 0
+	icon = 'icons/effects/mob_hunt.dmi'
+	var/state_name
+	var/datum/mob_hunt/mob_info = null
+	var/list/clients_encountered = list()		//tracks who has already interacted with us, so they can't attempt a second capture
+	var/image/avatar
+
+/obj/effect/nanomob/New(loc, datum/mob_hunt/new_info)
+	..()
+	if(!new_info)
+		qdel(src)
+		return
+	mob_info = new_info
+	update_self()
+	forceMove(mob_info.spawn_point)
+	if(!mob_info.is_trap)
+		addtimer(src, "despawn", mob_info.lifetime)
+
+/obj/effect/nanomob/proc/update_self()
+	if(!mob_info)
+		return
+	name = mob_info.mob_name
+	desc = "A wild [name] (level [mob_info.level]) appeared! Hit it with your PDA with the game open to attempt to capture it!"
+	if(mob_info.is_shiny)
+		state_name = mob_info.icon_state_shiny
+	else
+		state_name = mob_info.icon_state_normal
+	avatar = image(icon, src, state_name)
+	avatar.override = 1
+	add_alt_appearance("nanomob_avatar", avatar)
+
+/obj/effect/nanomob/attackby(obj/item/O, mob/user)
```

Comment:
Decided to leave this as the standard `obj/item/O` casting in case we want to add new items to interact with the nanomob effects later and so the declaration is more in line with the standard for the attackby proc's arguments